queue.go

```diff
@@ -0,0 +1,36 @@
+package queue
+
+import (
+	"fmt"
+)
+
+type Song string
+
+type Queue struct {
+	 songList map[Song]int
+}
+
+func NewQueue() *Queue {
+	q := Queue{
+		songList: make(map[Song]int),
+	}
+	return &q
+}
+
+func (q *Queue) AddSong(songID Song) error {
+	 song := q.songList[songID]
+	if song == 0 {

```

Comment:
This may be a dangerous game... You can check for membership in maps with the second return value, something like this: 
```
song, found := q.songList[songID]
if !found { ...
```